client/state/jetpack-connect/selectors.js

```diff
@@ -25,15 +27,18 @@ const getFlowType = function( state, site ) {
  * XMLRPC errors can be identified by the presence of an error message, the presence of an authorization code
  * and if the error message contains the string 'error'
  *
- * @param state
- * @returns {Boolean}
+ * @param {object} state Global state tree
+ * @returns {Boolean} If there's an xmlrpc error or not
  */
 const hasXmlrpcError = function( state ) {
+	const authorizeData = state.jetpackConnect.jetpackConnectAuthorize;
+
 	return (
-		state.authorizeError &&
-		state.authorizeError.message &&
-		state.authorizationCode &&
-		state.authorizeError.message.indexOf( 'error' ) > -1
+		authorizeData &&
```

Comment:
`authorizeData` will always be at least an empty object because of the reducer's initial state.

Also, for checking the existence of nested properties, Lodash's [`_.has`](https://lodash.com/docs/4.16.2#has) or [`_.get`](https://lodash.com/docs/4.16.2#get) may come in handy:

```js
const hasXmlrpcError = function( state ) {
	const authorizeData = state.jetpackConnect.jetpackConnectAuthorize;

	return (
		authorizeData.authorizationCode &&
		includes( get( authorizeData, [ 'authorizeError', 'message' ] ), 'error' )
	);
};
```